pkg/manifests/manifests.go

```diff
@@ -165,9 +170,44 @@ func (f *Factory) RouterDaemonSet(cr *ingressv1alpha1.ClusterIngress) (*appsv1.D
 
 	ds.Spec.Template.Spec.Containers[0].Env = append(ds.Spec.Template.Spec.Containers[0].Env, env...)
 
+	// Fill in the default certificate secret name.
+	secretName := fmt.Sprintf("router-certs-%s", cr.Name)
+	if cr.Spec.DefaultCertificateSecret != nil && len(*cr.Spec.DefaultCertificateSecret) > 0 {
+		secretName = *cr.Spec.DefaultCertificateSecret
+	}
+	ds.Spec.Template.Spec.Volumes[0].Secret.SecretName = secretName
+
 	return ds, nil
 }
 
+func (f *Factory) RouterServiceInternal(cr *ingressv1alpha1.ClusterIngress) (*corev1.Service, error) {
+	s, err := NewService(MustAssetReader(RouterServiceInternal))
+	if err != nil {
+		return nil, err
+	}
+
+	name := "router-" + cr.Name
```

Comment:
Does this conflict with [the "external" service](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/manifests/manifests.go#L177)?

Reply:
Yes it did. I fixed this locally when I was testing - just pushed the commit up.